File: Python Leetcode/Solution43.py
# 43. Multiply String

class Solution43:
    def multiply(self, num1: str, num2: str) -> str:

        # return f"{int(num1) * int(num2)}"
        
        n1,n2 = 0,0
        for i in num1:
            n1 = n1*10 + (ord(i)-48)
        for i in num2:
            n2 = n2*10 + (ord(i)-48)
        return f"{n1*n2}"

"""class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        a=b=0
        for i in num1:
            a = a*10+(ord(i)-48)
        for i in num2:
            b = b*10+(ord(i)-48)
        return str(a*b)
"""

File: Java Leetcode/Solution43.java
// 43. Multiply String

public class Solution43 {
    public String multiply(String num1, String num2) {
        if(num1.equals("0")||num2.equals("0"))
            return "0";
        int num[]=new int [num1.length()+num2.length()];
        for(int i=num1.length()-1;i>=0;i--)
        {
            for(int j=num2.length()-1;j>=0;j--)
            {
                num[i+j+1]+=(num1.charAt(i)-'0')*(num2.charAt(j)-'0');
                num[i+j]+=num[i+j+1]/10;
                num[i+j+1]%=10;
            }
        }
        int i=0;
        while(num.length>i&&num[i]==0)
            i++;
        String s="";
        
        while(i!=num.length)
        {
            s+=num[i];
            i++;
        }
        return s;
    }
}
